### PR TITLE
add rivgraph

### PR DESCRIPTION
**Pull request by jsta**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jsta** [comment]
Wow, the builds seem to have worked!? @jonschwenk @elbeejay Can you post a comment saying you're willing to be a maintainer?

**Comment by elbeejay** [comment]
I'm willing to be a maintainer as well. Thanks for setting this up @jsta

**Comment by jsta** [comment]
@conda-forge-admin ping conda-forge/help-python please.

File: recipes/rivgraph/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/veinsoftheearth/rivgraph/archive/v{{ version }}.tar.gz
+  sha256: 797b3dbba2a2f135a39a9cbaebdf59f9194fa48808211c5733dee56b510a505c
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
```
Comment (chrisburr) [inline recipes/rivgraph/meta.yaml:12]:
```suggestion
```

File: recipes/rivgraph/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - python >=3.7    
+    - geopandas
+    - scikit-image
+    - opencv
+    - networkx <2.7 # necessary for correct numpy ops
+    - shapely =1.7.1
+    - matplotlib-base
+    - fastdtw
+    - loguru
+
```
Comment (chrisburr) [inline recipes/rivgraph/meta.yaml:31]: Nominally these should only bee needed in `run` (unless the package is importing itself which shouldn't be needed).

```suggestion
```

Comment (jsta) [reply recipes/rivgraph/meta.yaml:31]: Omitting these from the host section was giving me build errors. Is self importing going on in https://github.com/VeinsOfTheEarth/RivGraph/blob/master/rivgraph/__init__.py#L3 ?
@elbeejay @jonschwenk

**Comment by jsta** [comment]
Thanks for the comments @chrisburr! I think I've addressed them all.

File: recipes/rivgraph/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/rivgraph/meta.yaml]: The issue is this line:

* https://github.com/VeinsOfTheEarth/RivGraph/blob/fb3baecd2a7a5c0e15b6771f1d757a59219bbf53/setup.py#L2

There are several ways of avoiding it which are summarised here:

* https://packaging.python.org/en/latest/guides/single-sourcing-package-version/

My personal favourite is `setuptools_scm` (+`importlib.metadata` in `__init__.py` to set the `__version__` variable).

Regardless it doesn't matter for conda-forge so LGTM 👍

**Pull request merged**